### DISCUSSION TITLE
"Refusing to apply an empty vschema to a non-empty existing vschema"

### DISCUSSION THREAD

**Discussion by markski1** [planetscale/discussion#596] [Q&A]
Earlier today, I enabled "Foreign Key Constraints" in the beta tab, but after some thought decided it might be better to not use that, so I disabled it. No problems.

However, trying to deploy a schema change today, I find myself with this error:

![imagen](https://github.com/planetscale/discussion/assets/22557859/5af37cd2-0249-4ed0-aef9-7ca6f5e40bc2)

Apparently, it's trying to apply the following VSchema change
![imagen](https://github.com/planetscale/discussion/assets/22557859/ef617d9d-b094-4695-9ca7-c2843b8f8d7b)

Which the production branch seems to have, despite me having disabled the feature:

![imagen](https://github.com/planetscale/discussion/assets/22557859/cc345057-34c2-41df-a5dc-a7c0419ce8bc)

Furtermore, the FCK option in beta is disabled now.
![imagen](https://github.com/planetscale/discussion/assets/22557859/d075d1a6-3b99-4ee0-8db7-96b22d7e0c80)

How should I proceed to get this thing completely disabled and out of my production branch's VSchema?

AGAIN: No changes were made that make use of foreign key constraints. I just enabled it planning to use it, then disabled it when I decided not to.

**Comment by markski1** [comment]
Apparently, disabling safe migrations in the production branch to attempt something just immediatly removed the foreignKeyMode line and now everything just works.

After re-enabling safe migrations, the deploy request was valid, it went through, and everyone lived happily ever after. Yep.

I'm still leaving this open though because I'd like to know why this happened. I assume it's my fault, but I'm not sure how exactly I caused this problem, and how toggling safe migrations off and back on solved it.

**Comment by mscoutermarsh** [comment]
Thanks for reporting @markski1. Shared with the team and they are looking into it.

**Comment by markski1** [comment]
Hi @mscoutermarsh ; is this confirmed to have been a bug?

**Reply by mscoutermarsh** [reply]
Yes, vschema is something we handle for you and this is intended to work automatically.
